pyiceberg/io/fsspec.py

```diff
@@ -158,6 +160,14 @@ def _gs(properties: Properties) -> AbstractFileSystem:
     # https://gcsfs.readthedocs.io/en/latest/api.html#gcsfs.core.GCSFileSystem
     from gcsfs import GCSFileSystem
 
+    if (endpoint := properties.get(GCS_ENDPOINT)) and GCS_SERVICE_HOST not in properties:
+        deprecated(
```

Comment:
Now we have the deprecation_message function, which could be used here.

Reply:
Yes, that's much nicer, thanks for adding that @ndrluis